addons/payment_asiapay/models/payment_transaction.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from werkzeug import urls
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+from .const import CURRENCY_MAPPING
+from odoo.addons.payment_asiapay import utils as payment_asiapay_utils
+from odoo.addons.payment_asiapay.controllers.main import AsiaPayController
+
+_logger = logging.getLogger(__name__)
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    def _get_specific_rendering_values(self, processing_values):
+        """Override of payment to return asiapay rendering values.
+
+        Note: self.ensure_one() from `_get_processing_values`
+
+        :param dict processing_values: The generic and specific processing values of the transaction
+        :return: The dict of acquirer-specific processing values
+        :rtype: dict
+        """
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != "asiapay":
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        order_ref = "{} {}".format(self.reference, datetime.now().strftime("%m/%d/%Y %H:%M:%S"))
+        rendering_values = {
+            "merchantId": self.acquirer_id.asiapay_merchant_id,
+            "amount": self.amount,
+            "orderRef": order_ref,
+            "currCode": CURRENCY_MAPPING.get(self.acquirer_id.asiapay_currency_id.name),
+            "mpsMode": "SCP",
+            "successUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "failUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "cancelUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "payType": "N",
+            "lang": payment_asiapay_utils.get_lang(self.env.lang),
+            "payMethod": "ALL",
```

Comment:
Could you try playing with this parameter and see if you can indeed select a single payment method? It's important for one of our future tasks.

Reply:
Yes it is possible. Below is the list. Do you want me to restrict it to one payment method only?

![image](https://user-images.githubusercontent.com/85471608/185538726-bfc78f47-6af0-4529-bd54-f47e34562994.png)
![image](https://user-images.githubusercontent.com/85471608/185538783-b741db60-9c0e-4c66-917a-b659f30fe9f3.png)
![image](https://user-images.githubusercontent.com/85471608/185538813-72c0a645-0373-4060-bd74-0f96eeeb1ba4.png)

Reply:
However, it only accept one payment method type. Which means we need to select the payment method first and pass it to Asiapay